Some compilers on Windows don't work

Windows 7 (64-bit)
Nim Compiler Version 1.2.0 [Windows: amd64]                                     

VCC can compile some programs, but can't compile the nim compiler:
C:\nim\compiler> nim c --cc:vcc -d:release -t:/GL -l:"/link /LTCG" nim.nim
Hint:  [Link]
...
LINK : warning LNK4044: unrecognized option '/F33554432'; ignored
LINK : fatal error LNK1181: cannot open input file '..\icons\nim-amd64-windows-v
cc.res'
/link needs to be used when passing link options, presumably because cl.exe is being invoked instead of link.exe.
Powershell seems to be used during compilation of each file and I think it is slowing compilation down a lot. Task manager says that CPU of a powershell process is around 27. Why not remove it , require the user to run Nim from within a Visual Studio command prompt, and just find cl.exe and link.exe on the path?

clang_cl can compile a debug build, but can't compile a release build of seemingly any program:
clang-cl: error: no such file or directory: '/G7'                               
This seems to happen on the first object file compiled. Maybe the command is -Z7?
Both VCC and clang_cl could compile with the /MD option to reduce binary size by linking dynamically with ucrt.
Clang: Clang: Can't seem to link anything.
C:\nim\examples> nim c --cc:clang hallo.nim
...
stdlib_io.nim.c.o : error LNK2019: unresolved external symbol nimAddInt referenced in function writeWindows__JO9bn8JE1R6MjKZqdvhBrWg
stdlib_system.nim.c.o : error LNK2001: unresolved external symbol nimAddInt
stdlib_system.nim.c.o : error LNK2019: unresolved external symbol nimMulInt referenced in function newSeq
clang: error: linker command failed with exit code 1120 (use -v to see invocation)
Note: I did all this from the x64 Native Tools Command Prompt for VS 2019 command prompt, it didn't seem to change anything though.

--- a/compiler/extccomp.nim
+++ b/compiler/extccomp.nim
@@ -143,8 +143,8 @@ compiler vcc:
   result = (
     name: "vcc",
     objExt: "obj",
-    optSpeed: " /Ogityb2 /G7 ",
-    optSize: " /O1 /G7 ",
+    optSpeed: " /Ogityb2 ",
+    optSize: " /O1 ",
     compilerExe: "cl",
     cppCompiler: "cl",
     compileTmpl: "/c$vccplatform $options $include /Fo$objfile $file",

--- a/lib/nimbase.h
+++ b/lib/nimbase.h
@@ -569,7 +569,7 @@ NIM_STATIC_ASSERT(sizeof(NI) == sizeof(void*) && NIM_INTBITS == sizeof(NI)*8, ""
 #define nimModInt(a, b, res) (((*res) = (a) % (b)), 0)
 #define nimModInt64(a, b, res) (((*res) = (a) % (b)), 0)
 
-#if !defined(_MSC_VER) && !defined(NIM_EmulateOverflowChecks)
+#if (!defined(_MSC_VER) || defined(__clang__)) && !defined(NIM_EmulateOverflowChecks)
   /* these exist because we cannot have .compilerProcs that are importc'ed
     by a different name */
 

